src/controller/CHIPDeviceController.cpp

```diff
@@ -926,7 +934,12 @@ void DeviceCommissioner::OnRendezvousComplete()
     mPairedDevices.Insert(device->GetDeviceId());
     mPairedDevicesUpdated = true;
 
-    PersistDevice(device);
+    // Newly paired devices should be stored synchronously because we cannot mark pairing complete
+    // until they are available in storage
+    PersistDevice(device, true);
```

Comment:
nit: boolean arguments are non-obvious in general. One option would be to `PersitDevice(device, true /* sync */)` however even that feels awkward.

I would suggest using an enum of `kPersistSync` and `kPersistAsync` so that the `PersistDeviceList(kPersistSync)` is immediately obvious and type safe.

Reply:
I will add an enum here. Thanks!